File: Lab - 02/Propuestos/01/src/main/java/example/CalculatorArea.java
package example;

public class CalculatorArea {

    // Calcula el área de un rectángulo
    public double calculateRectangleArea(String baseStr, String heightStr) throws IllegalArgumentException {
        double base = parseDouble(baseStr);
        double height = parseDouble(heightStr);

        // Validación de valores negativos o cero
        if (base <= 0 || height <= 0) {
            throw new IllegalArgumentException("La base y la altura deben ser valores positivos");
        }

        return base * height;
    }

    // Convierte String a double con validaciones
    private double parseDouble(String value) throws IllegalArgumentException {
        if (value == null || value.trim().isEmpty()) {
            throw new IllegalArgumentException("El valor no puede estar vacío");
        }

        // Reemplaza comas por puntos para manejar formatos numéricos diferentes
        String normalizedValue = value.trim().replace(',', '.');

        try {
            return Double.parseDouble(normalizedValue);
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException("'" + value + "' no es un número válido");
        }
    }
}

File: Lab - 02/Propuestos Python/Ej01/calculator_area.py
class CalculatorArea:
    def calculate_rectangle_area(self, base_str, height_str):
        try:
            base = self._parse_double(base_str)
            height = self._parse_double(height_str)

            if base <= 0 or height <= 0:
                return "Error: La base y la altura deben ser valores positivos."

            area = base * height

            base_formatted = self._format_number(base)
            height_formatted = self._format_number(height)
            area_formatted = self._format_number(area)

            return f"Rectángulo con base: {base_formatted}, altura: {height_formatted}\nÁrea calculada: {area_formatted}"

        except ValueError:
            return "Error: Ingrese valores numéricos válidos para la base y la altura."
        except Exception as e:
            return f"Error inesperado: {str(e)}"

    def _parse_double(self, value):
        if value is None or value.strip() == "":
            raise ValueError("El valor no puede estar vacío")

        normalized_value = value.strip().replace(',', '.')
        try:
            return float(normalized_value)
        except ValueError:
            raise ValueError(f"'{value}' no es un número válido")

    def _format_number(self, number):
        if number == int(number):
            return str(int(number))
        else:
            return str(number)
